src/main/java/org/folio/linked/data/service/resource/impl/ResourceEdgeServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package org.folio.linked.data.service.resource.impl;
+
+import org.folio.ld.dictionary.PredicateDictionary;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.linked.data.model.entity.ResourceEdge;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+@Service
+public class ResourceEdgeServiceImpl {
+
+  // If there are more conditions like this, move them to an inner class or enum
+  private static final Predicate<ResourceEdge> IS_ADMIN_METADATA = edge -> edge.getPredicate().getUri()
+    .equals(PredicateDictionary.ADMIN_METADATA.getUri());
+  private static final Predicate<ResourceEdge> SOME_OTHER_CONDITION = edge -> true; // Dummy condition
+
+  private Map<ResourceTypeDictionary, Predicate<ResourceEdge>> rules;
+
+  public ResourceEdgeServiceImpl() {
+    initializeRules();
+  }
+
+  public void copyOutgoingEdges(Resource from, Resource to) {
+    this.rules.entrySet()
```

Comment:
Idea:
What if instead of hard-coded list of resource types we do it dynamically?
We already have a way to find a mapper by resource type / predicate / request/response dto: 
org.folio.linked.data.mapper.dto.common.SingleResourceMapperImpl#getMapperUnit
So, might be we will copy every resource which doesn't have a mapper?

Reply:
Hi @PBobylev, There is one problem with this approach.

Let us assume the current graph is like this
<img width="371" alt="image" src="https://github.com/user-attachments/assets/2d602e6d-6e61-49b8-8751-f08e4df0cf5f">

If a cataloger open this work in editor, remove the connection to Person 1, change the creator & illustrator to Person2, and then save, the resulting graph will look like below.
<img width="357" alt="image" src="https://github.com/user-attachments/assets/df49c50b-fd0a-46ae-954a-b5a4bb7a48ec">

Problem - We will keep the "illustrator" connection as `org.folio.linked.data.mapper.dto.common.SingleResourceMapperImpl#getMapperUnit` will return an empty optional for the `http://bibfra.me/vocab/relation/illustrator` predicate.

One option is to add "isExcludedEdge" method as I did in this PR. I do not like it at all. Also, I think it is dangerous. We will have similar situations in future (eg: after implementing [MODLD-361](https://folio-org.atlassian.net/browse/MODLD-361)). If the developer forgets to update "isExcludedEdge" method with additional edges, the graph will get corrupted.

So, I prefer going back to the original approach where we control what edges has to be copied over. This way, we have better control on what edges to retain.

Thoughts?